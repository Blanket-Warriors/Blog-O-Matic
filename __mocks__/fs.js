export const mkdir = jest.fn((a, cb) => cb())
export const writeFile = jest.fn((a, b, cb) => cb())
export const existsSync = jest.fn(() => true)
export const readFileSync = jest.fn()
