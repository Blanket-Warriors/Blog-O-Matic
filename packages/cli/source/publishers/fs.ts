import { ncp } from "ncp"
import { resolve } from "path"
import { optimizeImages } from "./helpers/optimizeImages"


export async function fsPublisher(output: string) {
  const input = resolve(process.cwd(), "resources")
  await optimizeImages({ input, output })
  ncp(resolve(process.cwd(), "posts"), resolve(output, "posts"), {}, () => {
    console.log("Done.")
  })
}
