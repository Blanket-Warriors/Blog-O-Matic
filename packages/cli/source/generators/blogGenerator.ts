import { promisify } from "@civility/utilities"
import * as fs from "fs"
import { createPromptModule } from "inquirer"
import { join } from "path"
import { configTemplate, ignoreTemplate, privateTemplate } from "../templates"
import { basicQs, scpQs, sshQs } from "./blogQuestions"
const prompt = createPromptModule()
const homedir = require("os").homedir()
const mkdir = promisify(fs.mkdir)
const writeFile = promisify(fs.writeFile)


export interface IConfig {
  AUTHOR: string
  DATE_FORMAT: string
  TITLE: string
  SERVICE: string
}

export interface IPrivate {
  FILESYSTEM_LOCATION?: string
  PORT?: string
  SERVER_LOCATION?: string
  SSH_LOCATION?: string
  USE_SSH?: boolean
}


export async function blogGenerator() {
  console.log("\nðŸ˜³ðŸ˜³ðŸ¤–ðŸ˜³ WELCOME TO BLOG-O-MATIC! ðŸ˜³ðŸ˜³ðŸ¤–ðŸ˜³\n")

  const config = await prompt(basicQs) as IConfig
  const privateInfo: IPrivate | null = config.SERVICE === "scp"
    ? await prompt(scpQs) as IPrivate
    : null

  if (privateInfo && privateInfo.USE_SSH) {
    const LOCATION_PROMPTS = await prompt(sshQs) as { SSH_LOCATION: string }
    privateInfo.SSH_LOCATION = LOCATION_PROMPTS.SSH_LOCATION
  }

  try {
    await Promise.all([
      mkdir(config.TITLE),
      mkdir(join(homedir, ".blog-o-matic"), { recursive: true }),
    ])
  } catch (error) {
    if (error.code === "EEXIST") return console.warn("FAILED: A blog already exists here!")
  }

  await Promise.all([
    writeFile(join(config.TITLE, "blog.config.yml"), configTemplate(config)),
    writeFile(join(config.TITLE, ".blogignore"), ignoreTemplate()),
    mkdir(join(config.TITLE, "resources")),
    mkdir(join(config.TITLE, "posts")),
  ])

  if (privateInfo) {
    writeFile(join(homedir, ".blog-o-matic", `${config.TITLE}.yml`), privateTemplate(privateInfo))
  }

  console.log(`
    Congratulations! ðŸŽ‰ ðŸŽ‰ ðŸŽ‰
    Blog-o-Matic successfully created your edit folder!
    You can write whatever you want in here!
    (Or if you're lazy like us, you can run \`blog post\`)
  `)
}
