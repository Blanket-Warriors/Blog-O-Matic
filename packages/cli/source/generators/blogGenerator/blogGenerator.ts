import { promisify } from "@civility/utilities"
import * as fs from "fs"
import { createPromptModule } from "inquirer"
import { dump } from "js-yaml"
import { join } from "path"
import { IConfig } from "../../definitions"
import * as q from "../questions"


const prompt = createPromptModule()
const mkdir = promisify(fs.mkdir)
const writeFile = promisify(fs.writeFile)
const ignoreText = `build/
.DS_Store
.Spotlight-V100
.Trashes
._*\n
`

const publisherQuestions: { [key: string]: any } = {
  fs: [ q.blogOut ],
  s3: [ q.s3Creds, q.s3Bucket ],
}

export async function blogGenerator() {
  console.log("\nðŸ˜³ðŸ˜³ðŸ¤–ðŸ˜³ WELCOME TO BLOG-O-MATIC! ðŸ˜³ðŸ˜³ðŸ¤–ðŸ˜³\n")

  const config = await prompt([
    q.blogAuthor,
    q.blogTitle,
    q.blogPublisher,
  ]) as IConfig

  Object.assign(config, { version: "4.0.0" })

  if (config.publisher && publisherQuestions[config.publisher]) {
    Object.assign(config, {
      [config.publisher]: await prompt(publisherQuestions[config.publisher]),
    })
  }

  await mkdir(config.title)

  await Promise.all([
    writeFile(join(config.title, "blog.config.yml"), dump(config)),
    writeFile(join(config.title, ".blogignore"), ignoreText),
    mkdir(join(config.title, "images")),
    mkdir(join(config.title, "posts")),
  ])

  console.log(`
    Congratulations! ðŸŽ‰ ðŸŽ‰ ðŸŽ‰
    Blog-o-Matic successfully created your edit folder!
    You can write whatever you want in here!
    (Or if you're lazy like us, you can run \`blog post\`)
  `)
}
