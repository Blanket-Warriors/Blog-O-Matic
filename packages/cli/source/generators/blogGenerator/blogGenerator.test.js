jest.mock("fs")
jest.mock("inquirer")

import { mkdir, writeFile } from "fs"
import { blogGenerator } from "./blogGenerator"
import { load } from "js-yaml"


beforeAll(() => { console.log = jest.fn() })

test("Should call writeFile", async () => {
  await blogGenerator()
  const [ mkdir1, mkdir2, mkdir3 ] = mkdir.mock.calls
  const [ writeFile1, writeFile2 ] = writeFile.mock.calls


  expect(mkdir1[0]).toBe("my-blog-title")
  expect(mkdir2[0]).toBe("my-blog-title/images")
  expect(mkdir3[0]).toBe("my-blog-title/posts")
  expect(writeFile1[0]).toBe("my-blog-title/blog.config.yml")
  expect(load(writeFile1[1])).toEqual({
    author: "my-author",
    host: "127.0.0.1",
    out: "my-out",
    path: "my-path",
    port: 8080,
    publisher: "my-publisher",
    title: "my-blog-title",
    version: "4.0.0",
  })
  expect(writeFile2[0]).toBe("my-blog-title/.blogignore")
  expect(writeFile2[1]).toBe("build/\n.DS_Store\n.Spotlight-V100\n.Trashes\n._*\n\n")

  expect(console.log).toBeCalledTimes(2)
  const [ welcomeText, finishedText ] = console.log.mock.calls
  expect(welcomeText[0]).toMatchSnapshot()
  expect(finishedText[0]).toMatchSnapshot()
})
