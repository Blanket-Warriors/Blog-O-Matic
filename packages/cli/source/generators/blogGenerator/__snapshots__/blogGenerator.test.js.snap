// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should call writeFile 1`] = `
"
😳😳🤖😳 WELCOME TO BLOG-O-MATIC! 😳😳🤖😳
"
`;

exports[`Should call writeFile 2`] = `
"
    Congratulations! 🎉 🎉 🎉
    Blog-o-Matic successfully created your edit folder!
    You can write whatever you want in here!
    (Or if you're lazy like us, you can run \`blog post\`)
  "
`;
